src/app/wallets/update-pending-invoices.ts

```diff
@@ -238,3 +217,64 @@ const updatePendingInvoice = async ({
     return true
   })
 }
+
+const declineHeldInvoice = async ({
+  walletInvoice,
+  logger,
+}: {
+  walletInvoice: WalletInvoice
+  logger: Logger
+}): Promise<boolean | ApplicationError> => {
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const walletInvoicesRepo = WalletInvoicesRepository()
+
+  const { pubkey, paymentHash } = walletInvoice
+
+  const lnInvoiceLookup = await lndService.lookupInvoice({ pubkey, paymentHash })
+
+  const pendingInvoiceLogger = logger.child({
+    hash: paymentHash,
+    lnInvoiceLookup,
+    walletInvoice,
+    topic: "payment",
+    protocol: "lightning",
+    transactionType: "receipt",
+    onUs: false,
+  })
+
+  if (lnInvoiceLookup instanceof InvoiceNotFoundError) {
+    const isDeleted = await walletInvoicesRepo.deleteByPaymentHash(paymentHash)
+    if (isDeleted instanceof Error) {
+      pendingInvoiceLogger.error("impossible to delete WalletInvoice entry")
+      return isDeleted
+    }
+    return false
+  }
+  if (lnInvoiceLookup instanceof Error) return lnInvoiceLookup
+
+  if (!lnInvoiceLookup.isHeld) {
+    pendingInvoiceLogger.info({ lnInvoiceLookup }, "invoice has not been paid yet")
+    return false
+  }
+
+  let heldForMsg = ""
+  if (lnInvoiceLookup.heldAt) {
+    heldForMsg = `for ${elapsedSinceTimestamp(lnInvoiceLookup.heldAt)}s `
+  }
+  pendingInvoiceLogger.error(
+    { lnInvoiceLookup },
+    `invoice has been held ${heldForMsg}and is now been cancelled`,
+  )
```

Comment:
I think this is what me might want?

```suggestion
  if (lnInvoiceLookup.heldAt) {
    heldForMsg = `for ${elapsedSinceTimestamp(lnInvoiceLookup.heldAt)}s `
      pendingInvoiceLogger.error(
    { lnInvoiceLookup },
    `invoice has been held ${heldForMsg}and is now been cancelled`,
  )
  }
```

Reply:
Oh no, we want that error log to happen whether or not there is a `lnInvoiceLookup.heldAt` value (see [here](https://github.com/GaloyMoney/galoy/pull/1188/files#diff-f02445547836eff187be3a4b54e41d92b4223dd7c304847b688ef3016eec317fR237)).

This conditional is purely for constructing the error log message, which is why I formatted the message to work even if there isn't a value for `heldAt`

It would just look like:
- `invoice has been held and is now been cancelled` without a value
- `invoice has been held for 100s and is now been cancelled` with a value